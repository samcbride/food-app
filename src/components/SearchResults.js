import React from "react";
import PropTypes from "prop-types";
import Recipe from "./Recipe";
import "../styles/SearchResults.css";

const SearchResults = ({ searchResults }) => {
  return (
    <div className="search-results" data-testid="search-results">
      {searchResults &&
        searchResults.map((recipe) => (
          <Recipe
            key={recipe.recipe.url}
            title={recipe.recipe.label}
            image={recipe.recipe.image}
            url={recipe.recipe.url}
          />
        ))}
    </div>
  );
};

SearchResults.propTypes = {
  searchResults: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      image: PropTypes.string,
      url: PropTypes.string,
    })
  ).isRequired,
};

export default SearchResults;
