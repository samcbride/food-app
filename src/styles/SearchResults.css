.search-results {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-content: center;
  flex-wrap: wrap;
  width: 100%;
}
