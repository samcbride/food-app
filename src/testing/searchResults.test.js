import React from "react";
import { render } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = {
    searchResults: [],
  };
  

  it("renders correctly", () => {
    const { asFragment } = render(<SearchResults 
      searchResults={validProps.searchResults}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("renders the correct number of SearchResults instances", () => {
    const { getAllByTestId } = render(
      <SearchResults 
      searchResults={validProps.searchResults}
      />
    );
    expect(getAllByTestId("search-results")).toHaveLength(1);
  })
});