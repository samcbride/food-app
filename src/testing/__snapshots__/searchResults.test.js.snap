// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
    data-testid="search-results"
  />
</DocumentFragment>
`;
